flow/connectors/clickhouse/normalize.go

```diff
@@ -159,14 +159,27 @@ func generateCreateTableSQLForNormalizedTable(
 	}
 
 	var engine string
-	if tableMapping == nil {
+	tmEngine := protos.TableEngine_CH_ENGINE_REPLACING_MERGE_TREE
+	if tableMapping != nil {
+		tmEngine = tableMapping.Engine
+	}
+	switch tmEngine {
+	case protos.TableEngine_CH_ENGINE_REPLACING_MERGE_TREE:
 		engine = fmt.Sprintf("ReplacingMergeTree(%s)", peerdb_clickhouse.QuoteIdentifier(versionColName))
-	} else if tableMapping.Engine == protos.TableEngine_CH_ENGINE_MERGE_TREE {
+	case protos.TableEngine_CH_ENGINE_MERGE_TREE:
 		engine = "MergeTree()"
-	} else if tableMapping.Engine == protos.TableEngine_CH_ENGINE_NULL {
+	case protos.TableEngine_CH_ENGINE_REPLICATED_REPLACING_MERGE_TREE:
+		engine = fmt.Sprintf(
+			"ReplicatedReplacingMergeTree('/clickhouse/tables/{shard}/{database}/{table}','{replica}',%s)",
+			peerdb_clickhouse.QuoteIdentifier(versionColName),
+		)
+	case protos.TableEngine_CH_ENGINE_REPLICATED_MERGE_TREE:
+		engine = fmt.Sprintf(
+			"ReplicatedMergeTree('/clickhouse/tables/{shard}/{database}/{table}','{replica}',%s)",
```

Comment:
Table name should be interpolated rather than template, as this has issues with rename which resync will run into

How this works with resync in general needs to be evaluated